compiler/aarch64/codegen/UnaryEvaluator.cpp

```diff
@@ -215,12 +215,14 @@ TR::Register *OMR::ARM64::TreeEvaluator::bu2iEvaluator(TR::Node *node, TR::CodeG
    {
    TR::Node *child = node->getFirstChild();
 
-   if ((child->getOpCodeValue() == TR::bload || child->getOpCodeValue() == TR::bloadi)
+   if (child->getReferenceCount() == 1
+       && (child->getOpCodeValue() == TR::bload || child->getOpCodeValue() == TR::bloadi)
        && child->getRegister() == NULL)
       {
       // Use unsigned load
       TR::Register *trgReg = commonLoadEvaluator(child, TR::InstOpCode::ldrbimm, cg);
       node->setRegister(trgReg);
+      cg->decReferenceCount(child);
```

Comment:
`commonLoadEvaluator()` creates a MemoryReference from `child` and then calls `decNodeReferenceCounts()` on the MemoryReference before returning.  Is this `decReferenceCount()` here necessary?

Reply:
`OMR::ARM64::MemoryReference::decNodeReferenceCounts()` calls `cg->decReferenceCount()` on `_baseNode` and `_indexNode` of the `child` node, but not on the `child` itself.

Reply:
Right, for indirect loads we step past the node to the base child.